subgraphs/rocket-pool/src/updaters/usageMetrics.ts

```diff
@@ -0,0 +1,111 @@
+import { Address, ethereum } from "@graphprotocol/graph-ts";
+import { getOrCreateProtocol } from "../entities/protocol";
+import {
+  Account,
+  ActiveAccount,
+  UsageMetricsDailySnapshot,
+  UsageMetricsHourlySnapshot,
+} from "../../generated/schema";
+import { SECONDS_PER_DAY, SECONDS_PER_HOUR } from "../utils/constants";
+
+export function updateUsageMetrics(block: ethereum.Block, from: Address): void {
+  const accountId = from.toHexString();
+  let account = Account.load(accountId);
+  if (!account) {
+    account = new Account(accountId);
+    account.save();
+
+    const protocol = getOrCreateProtocol();
+    protocol.cumulativeUniqueUsers += 1;
+    protocol.save();
+  }
+
+  const protocol = getOrCreateProtocol();
+  const usageMetricsDailySnapshot = getOrCreateUsageMetricsDailySnapshot(block);
+  const usageMetricsHourlySnapshot =
+    getOrCreateUsageMetricsHourlySnapshot(block);
+
+  usageMetricsDailySnapshot.dailyTransactionCount += 1;
+  usageMetricsHourlySnapshot.hourlyTransactionCount += 1;
+
+  usageMetricsDailySnapshot.cumulativeUniqueUsers =
+    protocol.cumulativeUniqueUsers;
+  usageMetricsHourlySnapshot.cumulativeUniqueUsers =
+    protocol.cumulativeUniqueUsers;
+
+  const dailyActiveAccountId = (block.timestamp.toI64() / SECONDS_PER_DAY)
+    .toString()
+    .concat("-")
+    .concat(accountId);
+
+  let dailyActiveAccount = ActiveAccount.load(dailyActiveAccountId);
+  if (!dailyActiveAccount) {
+    dailyActiveAccount = new ActiveAccount(dailyActiveAccountId);
+    dailyActiveAccount.save();
+
+    usageMetricsDailySnapshot.dailyActiveUsers += 1;
+    usageMetricsHourlySnapshot.hourlyActiveUsers += 1;
+  }
+
+  usageMetricsDailySnapshot.blockNumber = block.number;
+  usageMetricsHourlySnapshot.blockNumber = block.number;
+
+  usageMetricsDailySnapshot.timestamp = block.timestamp;
+  usageMetricsHourlySnapshot.timestamp = block.timestamp;
+
+  usageMetricsDailySnapshot.save();
+  usageMetricsHourlySnapshot.save();
+}
+
+export function getOrCreateUsageMetricsDailySnapshot(
+  block: ethereum.Block
+): UsageMetricsDailySnapshot {
+  const dayId: string = (block.timestamp.toI64() / SECONDS_PER_DAY).toString();
+  let usageMetrics = UsageMetricsDailySnapshot.load(dayId);
+
+  if (!usageMetrics) {
+    const protocol = getOrCreateProtocol();
+    usageMetrics = new UsageMetricsDailySnapshot(dayId);
+
+    usageMetrics.protocol = protocol.id;
+    usageMetrics.totalPoolCount = protocol.totalPoolCount;
+
+    usageMetrics.dailyActiveUsers = 0;
+    usageMetrics.cumulativeUniqueUsers = 0;
+    usageMetrics.dailyTransactionCount = 0;
+
+    usageMetrics.blockNumber = block.number;
```

Comment:
maybe good to have these on the outside of the if block like other getters

Reply:
fixed